presto-main/src/main/java/com/facebook/presto/operator/ChannelSet.java

```diff
@@ -76,13 +77,15 @@ public boolean contains(int position, Page page)
         private final GroupByHash hash;
         private final OperatorContext operatorContext;
         private final Page nullBlockPage;
+        private final LocalMemoryContext localMemoryContext;
 
         public ChannelSetBuilder(Type type, Optional<Integer> hashChannel, int expectedPositions, OperatorContext operatorContext, JoinCompiler joinCompiler)
         {
             List<Type> types = ImmutableList.of(type);
             this.hash = createGroupByHash(operatorContext.getSession(), types, HASH_CHANNELS, hashChannel, expectedPositions, joinCompiler);
             this.operatorContext = operatorContext;
             this.nullBlockPage = new Page(type.createBlockBuilder(new BlockBuilderStatus(), 1, UNKNOWN.getFixedSize()).appendNull().build());
+            this.localMemoryContext = operatorContext.localUserMemoryContext();
```

Comment:
Looking at the original implementation, it appears like `operatorContext` may be null (`if (operatorContext != null)` below). I assume this code will crash with NPE in such a case.

If it turns out `operatorContext` is guaranteed to be non-null, can you add rquireNonNull to `operatorContext`?

Reply:
The null check in the original implementation is totally confusing, because the constructor already accesses `operatorContext` on L85. We should add a non-null check in the constructor.